p2p/module_raintree_test.go

```diff
@@ -367,85 +363,75 @@ func prepareConnMock(t *testing.T, expectedNumNetworkReads, expectedNumNetworkWr
 		require.NoError(t, err)
 		p2pModules[validatorId(t, i+1)] = p2pMod.(*p2pModule)
 	}
 	return
 }
 
-func createTestingGenesisAndConfigFiles(t *testing.T, cfg modules.Config, genesisState modules.GenesisState, n int) {
-	config, err := json.Marshal(cfg.P2P)
-	require.NoError(t, err)
-
-	genesis, err := json.Marshal(genesisState.ConsensusGenesisState)
-	require.NoError(t, err)
-
-	genesisFile := make(map[string]json.RawMessage)
-	configFile := make(map[string]json.RawMessage)
-	moduleName := new(p2pModule).GetModuleName()
-
-	genesisFile[test_artifacts.GetGenesisFileName(moduleName)] = genesis
-	configFile[moduleName] = config
-	genesisFileBz, err := json.MarshalIndent(genesisFile, "", "    ")
-	require.NoError(t, err)
-
-	p2pFileBz, err := json.MarshalIndent(configFile, "", "    ")
-	require.NoError(t, err)
-	require.NoError(t, ioutil.WriteFile(testingGenesisFilePath+jsonPosfix, genesisFileBz, 0777))
-	require.NoError(t, ioutil.WriteFile(testingConfigFilePath+strconv.Itoa(n)+jsonPosfix, p2pFileBz, 0777))
-}
-
-func createConfigs(t *testing.T, numValidators int) (configs []modules.Config, genesisState modules.GenesisState) {
-	configs = make([]modules.Config, numValidators)
+func createMockRuntimeMgrs(t *testing.T, numValidators int) []modules.RuntimeMgr {
+	ctrl := gomock.NewController(t)
+	mockRuntimeMgrs := make([]modules.RuntimeMgr, numValidators)
 	valKeys := make([]cryptoPocket.PrivateKey, numValidators)
 	copy(valKeys[:], keys[:numValidators])
-	genesisState = createGenesisState(t, valKeys)
-	for i := range configs {
-		configs[i] = modules.Config{
-			Base: &modules.BaseConfig{
-				RootDirectory: "",
-				PrivateKey:    valKeys[i].String(),
-			},
-			P2P: &typesP2P.P2PConfig{
-				PrivateKey:            valKeys[i].String(),
-				ConsensusPort:         8080,
-				UseRainTree:           true,
-				IsEmptyConnectionType: true,
-			},
-		}
+	mockGenesisState := createMockGenesisState(t, valKeys)
+	for i := range mockRuntimeMgrs {
+		mockConfig := modulesMock.NewMockConfig(ctrl)
+		mockConfig.EXPECT().GetBaseConfig().Return(&runtime.BaseConfig{
+			RootDirectory: "",
+			PrivateKey:    valKeys[i].String(),
+		}).AnyTimes()
+		mockConfig.EXPECT().GetP2PConfig().Return(&typesP2P.P2PConfig{
+			PrivateKey:            valKeys[i].String(),
+			ConsensusPort:         8080,
+			UseRainTree:           true,
+			IsEmptyConnectionType: true,
+		}).AnyTimes()
+
+		mockRuntimeMgr := modulesMock.NewMockRuntimeMgr(ctrl)
+		mockRuntimeMgr.EXPECT().GetConfig().Return(mockConfig).AnyTimes()
+		mockRuntimeMgr.EXPECT().GetGenesis().Return(mockGenesisState).AnyTimes()
+		mockRuntimeMgrs[i] = mockRuntimeMgr
 	}
-	return
+	return mockRuntimeMgrs
 }
 
 func validatorId(_ *testing.T, i int) string {
 	return fmt.Sprintf(serviceUrlFormat, i)
 }
 
-func createGenesisState(t *testing.T, valKeys []cryptoPocket.PrivateKey) modules.GenesisState {
+func createMockGenesisState(t *testing.T, valKeys []cryptoPocket.PrivateKey) modules.GenesisState {
```

Comment:
Blocker:

Please add comment around any Mock creation, see the other mocks in this file for example around expected behaviors